src/main/java/org/gnode/nix/DataArray.java

```diff
@@ -646,6 +649,218 @@ public void setPolynomCoefficients(double[] polynomCoefficients) {
+                                      @Cast("const void*") float[] data,
+                                      @Const @ByRef NDSize count,
+                                      @Const @ByRef NDSize offset);
+
+    private native void setDataDirect(@Cast("nix::DataType") int dtype,
+                                      @Cast("const void*") double[] data,
+                                      @Const @ByRef NDSize count,
+                                      @Const @ByRef NDSize offset);
+
+    //--------------------------------------------------
+    // public getData methods
+    //--------------------------------------------------
+
+    /**
+     * Get stored data as byte array.
+     *
+     * @param count  dimensions
+     * @param offset offset
+     * @return byte array
+     */
+    public byte[] getDataAsByteArray(NDSize count, NDSize offset) {
+        byte[] data = new byte[(int) count.getElementsProduct()];
+        getDataDirect(DataType.Int8, data, count, offset);
+        return data;
+    }
+
+    /**
+     * Get stored data as short array.
+     *
+     * @param count  dimensions
+     * @param offset offset
+     * @return short array
+     */
+    public short[] getDataAsShortArray(NDSize count, NDSize offset) {
+        short[] data = new short[(int) count.getElementsProduct()];
+        getDataDirect(DataType.Int16, data, count, offset);
+        return data;
+    }
+
+    /**
+     * Get stored data as int array.
+     *
+     * @param count  dimensions
+     * @param offset offset
+     * @return int array
+     */
+    public int[] getDataAsIntArray(NDSize count, NDSize offset) {
+        int[] data = new int[(int) count.getElementsProduct()];
+        getDataDirect(DataType.Int32, data, count, offset);
+        return data;
+    }
+
+    /**
+     * Get stored data as long array.
+     *
+     * @param count  dimensions
+     * @param offset offset
+     * @return long array
+     */
+    public long[] getDataAsLongArray(NDSize count, NDSize offset) {
+        long[] data = new long[(int) count.getElementsProduct()];
+        getDataDirect(DataType.Int64, data, count, offset);
+        return data;
+    }
+
+    /**
+     * Get stored data as float array.
+     *
+     * @param count  dimensions
+     * @param offset offset
+     * @return float array
+     */
+    public float[] getDataAsFloatArray(NDSize count, NDSize offset) {
+        float[] data = new float[(int) count.getElementsProduct()];
+        getDataDirect(DataType.Float, data, count, offset);
+        return data;
+    }
+
+    /**
+     * Get stored data as double array.
+     *
+     * @param count  dimensions
+     * @param offset offset
+     * @return double array
+     */
+    public double[] getDataAsDoubleArray(NDSize count, NDSize offset) {
+        double[] data = new double[(int) count.getElementsProduct()];
+        getDataDirect(DataType.Double, data, count, offset);
+        return data;
+    }
+
```

Comment:
I would prefer if the getters would resemble more the setters and have the array as an input parameter:

```java
public void getData(int[] data, NDSize count, NDSize offset)

public void getData(int[] data, NDSize count, NDSize offset, 
                    int dataType)

// same for other primitive types
```